modules/gdscript/gdscript_analyzer.cpp

```diff
@@ -1089,12 +1089,28 @@ void GDScriptAnalyzer::resolve_for(GDScriptParser::ForNode *p_for) {
 		}
 	}
 
-	if (!list_resolved) {
+	GDScriptParser::DataType variable_type;
+	if (list_resolved) {
+		variable_type.type_source = GDScriptParser::DataType::ANNOTATED_INFERRED;
+		variable_type.kind = GDScriptParser::DataType::BUILTIN;
+		variable_type.builtin_type = Variant::INT; // Can this ever be a float or something else?
```

Comment:
I guess it only reaches this is `range()` is used which in practice can only be an array of ´int´s. Maybe we can can change `range()` to return a PackedInt64Array instead and then just get the type from there to avoid mismatches. This can be done in a later change though.

Reply:
I'll take your offer to leave it as it is now if you think it's already any good and let you (or me in a future) enhance it.